### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by keda-vlad**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@RequestMapping("/auth")
+@Tag(name = "Authentication management", description = "Endpoint for login and registration")
+@RequiredArgsConstructor
+public class AuthenticationController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/login")
+    @Operation(summary = "login user", description = "login user")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto request) {
```
Comment (ivan0dyatlyukkk) [inline src/main/java/mate/academy/bookstore/controller/AuthenticationController.java]: I think that you need to add @Valid annotation

File: src/main/java/mate/academy/bookstore/dto/book/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.bookstore.dto.book;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+import org.hibernate.validator.constraints.ISBN;
+
+public record CreateBookRequestDto(
+        @NotNull String title,
+        @NotNull String author,
```
Comment (ivan0dyatlyukkk) [inline src/main/java/mate/academy/bookstore/dto/book/CreateBookRequestDto.java]: What will happen if we enter title: " " and author: " " ? @NotNull annotation doesn't prevent the case when we enter just blank string

File: src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        User user = new User();
+        user.setPassword(passwordEncoder.encode(request.password()));
+        user.setFirstName(request.firstName());
+        user.setLastName(request.lastName());
+        user.setEmail(request.email());
+        Role role = roleService.getByName(Role.RoleName.ROLE_USER);
+        user.setRoles(Set.of(role));
+        return userMapper.toResponseDto(userRepository.save(user));
+
```
Comment (ivan0dyatlyukkk) [inline src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java]: Remove the redundant line

File: src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java [hunk 4]
```diff
@@ -1,13 +1,8 @@
...
 import org.springframework.boot.test.context.SpringBootTest;
 
 @SpringBootTest
 class BookstoreApplicationTests {
 
-    @Test
-    void contextLoads() {
-    }
-
 }
```
Comment (ivan0dyatlyukkk) [inline src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java:8]: Why did you delete it?:)

File: src/main/java/mate/academy/bookstore/model/User.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+@Entity
+@Table(name = "users")
+@SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted = false")
+@Data
+public class User implements UserDetails {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Email
```
Comment (ivan0dyatlyukkk) [inline src/main/java/mate/academy/bookstore/model/User.java:33]: I guess it is not necessary to use the @Email

**Review by ivan0dyatlyukkk — CHANGES_REQUESTED** [review]

File: src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java (hunk 4 above)
Comment (keda-vlad) [reply src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java:8]: Why do I need it?

Comment (ivan0dyatlyukkk) [reply src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java:8]: As Bogdan told, this method runs the whole application for testing.
Empty contextLoads() is a test to verify if the application is able to load Spring context successfully or not.

Comment (keda-vlad) [reply src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java:8]: Firstly i don't have any of tests yet. Secondly, if app can't load spring context, it can't work at all, and i don't think that i can miss that. And at the end it starts. 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] 
[INFO] Results:
[INFO]
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO]

**Review by ivan0dyatlyukkk — APPROVED** [review]
Approved! Let's ask mentors about the method :)

File: src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java (hunk 4 above)
Comment (ivan0dyatlyukkk) [reply src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java:8]: I was just interested in that if your app worked with that method :)

Comment (ivan0dyatlyukkk) [reply src/test/java/mate/academy/bookstore/BookstoreApplicationTests.java:8]: I guess that you didn't add jwt.expiration and jwt.secret to application.properties in test folder (I can't find any commits with these changes), therefore you faced the issue with contextLoads test method

File: src/main/java/mate/academy/bookstore/model/User.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    private String lastName;
+    @ManyToMany
+    @JoinTable(
+            name = "user_role",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    private Set<Role> roles;
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/model/User.java]: It's better to assign collection implementation here so Hibernate know which proxied collection impl to choose

**Review by vovasalyha — APPROVED** [review]
